File: templates/javascript/v1/bridge/notifiers.js
export class Notifier {
  constructor(formatter) {
    this.formatter = formatter;
  }

  notify(message) {
    throw new Error("Not implemented");
  }
}

export class EmailNotifier extends Notifier {
  notify(message) {
    return this.formatter.format("email", message);
  }
}

export class SmsNotifier extends Notifier {
  notify(message) {
    return this.formatter.format("sms", message);
  }
}

File: templates/python/v1/bridge/notifiers.py
from __future__ import annotations

from dataclasses import dataclass

from bridge.formatters import Formatter


@dataclass(slots=True)
class Notifier:
    formatter: Formatter

    def notify(self, message: str) -> str:
        raise NotImplementedError


@dataclass(slots=True)
class EmailNotifier(Notifier):
    def notify(self, message: str) -> str:
        return self.formatter.format("email", message)


@dataclass(slots=True)
class SmsNotifier(Notifier):
    def notify(self, message: str) -> str:
        return self.formatter.format("sms", message)
